src/algebra/add_comm_torsor.lean

```diff
@@ -0,0 +1,184 @@
+end
+
+/-- Adding a vector to a point, then subtracting another point,
+produces the same result as subtracting the points then adding the
+vector. -/
+lemma vadd_vsub_comm (p1 p2 : P) (v : V) : p1 +ᵥ v -ᵥ p2 = (p1 -ᵥ p2) + v :=
+begin
+  apply vadd_cancel_left V p2,
+  rw [vadd_vsub_self V p2, ←vadd_add_assoc V p2, vadd_vsub_self V p2]
+end
+
+/-- Subtracting the result of adding a vector produces the same result
+as subtracting the points and subtracting that vector. -/
+lemma vsub_vadd_eq_vsub_sub (p1 p2 : P) (v : V) : p1 -ᵥ (p2 +ᵥ v) = (p1 -ᵥ p2) - v :=
+begin
+  apply vadd_cancel_left V (p2 +ᵥ v),
+  rw [vadd_vsub_self V (p2 +ᵥ v), vadd_comm V p2, vadd_add_assoc V p2, sub_add_cancel,
+      vadd_vsub_self V p2]
+end
+
+/-- Subtracting a point from itself produces 0. -/
+@[simp] lemma vsub_self (p : P) : p -ᵥ p = (0 : V) :=
+by rw [←add_zero (p -ᵥ p : V), ←vadd_vsub_comm V p, vsub_vadd_self V p]
+
+/-- If subtracting two points produces 0, they are equal. -/
+lemma eq_of_vsub_eq_zero {p1 p2 : P} (h : p1 -ᵥ p2 = (0 : V)) : p1 = p2 :=
+by rw [←vadd_vsub_self V p2 p1, h, vadd_zero V p2]
+
+/-- Subtracting two points produces 0 if and only if they are
+equal. -/
+lemma vsub_eq_zero_iff_eq (p1 p2 : P) : p1 -ᵥ p2 = (0 : V) ↔ p1 = p2 :=
+iff.intro (eq_of_vsub_eq_zero V) (λ h, h ▸ vsub_self V _)
+
+/-- Subtracting two points in the reverse order produces the negation
+of subtracting them. -/
+lemma vsub_rev_eq_neg_vsub (p1 p2 : P) : (p2 -ᵥ p1 : V) = -(p1 -ᵥ p2) :=
+begin
+  symmetry,
+  apply neg_eq_of_add_eq_zero,
+  apply vadd_cancel_left V p2,
+  rw [vadd_zero V p2, ←vadd_add_assoc V p2, vadd_vsub_self V p2, vadd_vsub_self V p1]
+end
+
+/-- If the same vector added to two points produces equal results,
+those points are equal. -/
+lemma vadd_cancel_right (p1 p2 : P) (v : V) (h : p1 +ᵥ v = p2 +ᵥ v) : p1 = p2 :=
+begin
+  have h2 : p1 +ᵥ v +ᵥ -v = p2 +ᵥ v +ᵥ -v, { rw h },
+  rwa [vadd_add_assoc V p1 v (-v), vadd_add_assoc V p2 v (-v), add_right_neg,
+      vadd_zero V p1, vadd_zero V p2] at h2
+end
+
+/-- Cancellation adding the results of two subtractions. -/
+@[simp] lemma add_vsub_vsub_cancel (p1 p2 p3 : P) : (p1 -ᵥ p2 : V) + (p2 -ᵥ p3) = (p1 -ᵥ p3) :=
+begin
+  apply vadd_cancel_left V p3,
+  rw [add_comm, ←vadd_add_assoc V p3, vadd_vsub_self V p3, vadd_vsub_self V p2,
+      vadd_vsub_self V p3]
+end
+
+/-- Cancellation subtracting the results of two subtractions. -/
+@[simp] lemma sub_vsub_vsub_cancel_right (p1 p2 p3 : P) :
+  (p1 -ᵥ p3 : V) - (p2 -ᵥ p3) = (p1 -ᵥ p2) :=
+by rw [←vsub_vadd_eq_vsub_sub V p1 p3, vadd_vsub_self V p3]
+
+/-- Cancellation subtracting the results of two subtractions. -/
+@[simp] lemma sub_vsub_vsub_cancel_left (p1 p2 p3 : P) :
+  (p3 -ᵥ p2 : V) - (p3 -ᵥ p1) = (p1 -ᵥ p2) :=
+by rw [sub_eq_add_neg, ←vsub_rev_eq_neg_vsub, add_comm, add_vsub_vsub_cancel]
+
+/-- The pairwise differences of a set of points. -/
+def vsub_set (s : set P) : set V := ⋃₀((λ p1 : P, (λ p2 : P, p1 -ᵥ p2) '' s) '' s)
```

Comment:
I think that it makes sense to use the same style as in `algebra/pointwise`.
```lean
def pointwise_mul [has_mul α] : has_mul (set α) :=
  ⟨λ s t, {a | ∃ x ∈ s, ∃ y ∈ t, a = x * y}⟩
```

Reply:
I've made that change.